### ISSUE TITLE
Add DSL.timestampDiff(Field, Field, DatePart) to generate the difference between timestamps in days, hours, minutes, etc.

### ISSUE THREAD

**Issue by sytolk** [jOOQ/jOOQ#4303]
Hi,
Im trying to generate with Jooq this SQL (MySQL):

```
AND TIMESTAMPDIFF(MINUTE,region_arrive_time,NOW())>region_wait_threshold
```

but the problem is that timestampDiff return DayToSecond only but REGION_WAIT_THRESHOLD field have Minutes (Integer field)

``` java
.and(DSL.timestampDiff(CARS.REGION_ARRIVE_TIME,new Timestamp(new Date().getTime())).ge(CARS.REGION_WAIT_THRESHOLD))
```

Database support MINUTE parameter in TIMESTAMPDIFF function but Jooq`s timestampDiff don`t have this parameter.

**Comment by sytolk** [comment]
I have change it like this but I think it will be better to have parameter Minutes Seconds.. in function timestampDiff

```java
.and(DSL.timestampDiff(CARS.REGION_ARRIVE_TIME, DSL.currentTimestamp().sub(CARS.REGION_WAIT_THRESHOLD.mul(60000))).ge(new DayToSecond(0)))
```

**Comment by lukaseder** [comment]
Thanks for reporting this. Yes, it's true. Thus far, we haven't implemented alternatives where `DSL.timestampDiff()` also takes a `DatePart` argument. In case such funcitonality is missing for you, you can always make it available immediately by creating a helper function like this:

```java
public static Field<Integer> timestampDiff(DatePart part, Field<Timestamp> t1, Field<Timestamp> t2) {
    return DSL.field("timestampdiff({0}, {1}, {2})", Integer.class, DSL.keyword(part.toSQL()), t1, t2);
}
```

See also the manual's section about "plain SQL" for more details:
http://www.jooq.org/doc/latest/manual/sql-building/plain-sql/

**Comment by lukaseder** [comment]
This has been implemented in jOOQ 3.15.0 as part of #7900

**Comment by lukaseder** [comment]
Duplicate of #7900
